### ISSUE TITLE
Assignment operator and no-constant-condition

### ISSUE THREAD

**Issue by zloirock** [eslint/eslint#2155]
Why assignment operator in condition trigger `no-constant-condition`? Maybe it's bad practice, but not constant:
```js
function clz32(x){
  return (x >>>= 0) ? 32 - x.toString(2).length : 32; // trigger no-constant-condition
}

var x = 6;
while(x -= 2)console.log(x);  // trigger no-constant-condition
```

**Comment by nzakas** [comment]
Can you please include the ESLint version you're using as well as pasting the actual console output?

**Comment by zloirock** [comment]
0.17.1, `2:0   error  Unexpected constant condition         no-constant-condition` for both examples.

**Comment by nzakas** [comment]
Ah, looks like Assignment Expression only checks the right expression: 
https://github.com/eslint/eslint/blob/master/lib/rules/no-constant-condition.js#L38

Definitely a bug.

**Comment by nzakas** [comment]
Working on this.
